enemy.py

```diff
@@ -0,0 +1,48 @@
+import window as w
+import constants
+import pygame
+import spaceship
+import random
+import enemy_tools
+
+
+class Enemy:
+    def __init__(self,enemy_type, is_boss, stage):
+        self.enemy_type = enemy_type
+        self.health = constants.health[enemy_type]
+        self.speed = random.choice([-1, 1]) * 2
+        self.mouth_state = 'closed'
+        self._last_shot = pygame.time.get_ticks()
+        self.width = enemy_tools.get_enemy_size_from_type(enemy_type)
+        self.height = enemy_tools.get_enemy_size_from_type(enemy_type)
+        self._x = random.randint(0, w.WIDTH - self.width)
+        self.y = 50 if stage >= 3 else random.randint(50, 150)
+        self.is_boss = is_boss
```

Comment:
A class is like a blueprint for a house. An "instance" is an actual house itself.

Here, we have a class for an "Enemy" but it's not an enemy _itself_. It's just the blueprint for how to define one. 

We create an actual enemy later in this file.